core/domain/state_domain.py

```diff
@@ -149,6 +149,28 @@ def validate(self, interaction, exp_param_specs_dict):
 
         self.outcome.validate()
 
+    @classmethod
+    def convert_html_in_answer_group(cls, answer_group_dict, conversion_fn):
+        """Checks for HTML fields in an answer group and converts it according
+        to the conversion function.
+
+        Args:
+            answer_group_dict: dict. The answer group dict.
+            conversion_fn: function. The function to be used for converting the
+                HTML.
+        Returns:
+            answer_group_dict. dict. The converted answer group dict.
+        """
+        answer_group_dict['outcome']['feedback']['html'] = (
+            conversion_fn(answer_group_dict['outcome']['feedback']['html']))
+        for rule_spec_index, rule_spec in enumerate(
+                answer_group_dict['rule_specs']):
+            rule_spec_object = RuleSpec.from_dict(rule_spec)
+            answer_group_dict['rule_specs'][rule_spec_index] = (
+                rule_spec_object.convert_html_in_rule_spec(
```

Comment:
This is confusing, will this method return a dict? this seems like a property method that accepts a dict and returns a dict? In that case why it's not a class method?

Reply:
done, changed it